Assets/NuGet/Editor/NugetHelper.cs

```diff
@@ -1291,5 +1302,119 @@ public static Texture2D DownloadImage(string url)
+                {
+                    credentialProviderPath = possiblePathWithFilename;
+                    break;
+                }
+            }
+
+            // If the credential provider was not found then download it.
+            if (String.IsNullOrEmpty(credentialProviderPath))
+            {
+                // Fallback to looking in the root of the Packages folder for the credential provider, and if it's not there download it
+                credentialProviderPath = Path.Combine(NugetConfigFile.RepositoryPath, credentialProviderFilename);
+
+                HttpWebRequest credentialProviderRequest = (HttpWebRequest)WebRequest.Create("https://" + packageHost + "/_apis/public/nuget/client/CredentialProviderBundle.zip");
+                Stream credentialProviderDownloadStream = credentialProviderRequest.GetResponse().GetResponseStream();
+
+                using (Stream file = File.Create(credentialProviderBundleFilename))
+                {
+                    CopyStream(credentialProviderDownloadStream, file);
+                }
+
+                // Unzip the bundle and extract the credential provider executable
+                using (ZipFile zip = ZipFile.Read(credentialProviderBundleFilename))
+                {
+                    foreach (ZipEntry entry in zip)
+                    {
+                        if (entry.FileName.ToLower() == credentialProviderFilename)
+                        {
+                            entry.Extract(NugetConfigFile.RepositoryPath, ExtractExistingFileAction.OverwriteSilently);
+                        }
+                    }
+                }
+
+                // Delete the bundle
+                File.Delete(credentialProviderBundleFilename);
+            }
+
+            // Launch the credential provider executable and get the json encoded response from the std output
+            Process process = new Process();
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.CreateNoWindow = true;
+            process.StartInfo.RedirectStandardOutput = true;
+            process.StartInfo.FileName = credentialProviderPath;
+            process.StartInfo.Arguments = "-uri " + "https://" + packageHost;
+
+            // http://stackoverflow.com/questions/16803748/how-to-decode-cmd-output-correctly
+            // Default = 65533, ASCII = ?, Unicode = nothing works at all, UTF-8 = 65533, UTF-7 = 242 = WORKS!, UTF-32 = nothing works at all
+            process.StartInfo.StandardOutputEncoding = Encoding.GetEncoding(850);
+            process.Start();
+            process.WaitForExit();
+
+            string output = process.StandardOutput.ReadToEnd();
+            if ((CredentialProviderExitCode)process.ExitCode == CredentialProviderExitCode.Success && !string.IsNullOrEmpty(output))
+            {
+                CredentialProviderResponse response = JsonUtility.FromJson<CredentialProviderResponse>(output);
```

Comment:
Hey! Working well from within the main project, but for packaging the CreateDLL project has to build against the version used by the packager - 5.2.0f3. Unfortunately I'm seeing this from that version:
```
 error CS0103: The name 'JsonUtility' does not exist in the current context
```
We've got a build script internally that builds unity packages from this repo. It wasn't included originally because UnitySetup wasn't OSS at the time. I think now I'll make a PR to add that script to this repo so it's easier to test.

Reply:
@GlitchEnzo - how do you feel about moving up our minimum supported version? Any idea how many people are still in the 5.x?

Reply:
It depends upon how far up we'd be moving the minimum.  The last metrics I saw (which are quite old) show a very low 5.2 usage.  I do know of a "major" customer that is still using 5.6, though. Some of the big companies are much slower in their upgrading.

The current version is still available to folks, so I won't push back too hard if you need to upgrade to an even higher minimum to help keep the development easier. If they desperately need old Unity support, they can always fork it themselves.